### PR TITLE
Implements the new test preparation service

### PR DESCRIPTION
**Pull request by apozsuse**
### What does this PR do? Why are we making this change?

This PR is part of the improvement effort in the test strategy. This PR implements the `test_preparation` service, which is the service that is executed right before the test service. The main goal for this test_preparation service is to copy the created image into the test_regions configured in the database for the account, so some tests (with different instance types/configurations) can be executed in these additional regions.

### How will these changes be tested?

The tests have been tested in a local development account against my aws personal account.

### How will this change be deployed? Any special considerations?
Nope

### Additional Information

This new service is really very similar to the replicate service.
Unit tests have been included for the service.

### CODE REVIEW

**Review by smarlowucf — APPROVED** [review]
I wonder if there's a way for mash to just redirect to the replication service instead of having a brand new service? If it's basically the same functions... 🤔

**Comment by apozsuse** [comment]
> I wonder if there's a way for mash to just redirect to the replication service instead of having a brand new service? If it's basically the same functions... 🤔



> I wonder if there's a way for mash to just redirect to the replication service instead of having a brand new service? If it's basically the same functions... 🤔



> I wonder if there's a way for mash to just redirect to the replication service instead of having a brand new service? If it's basically the same functions... 🤔

There's an option to switch the order of the replicate service and put it after create, but the clean up if something fails would be more complex.

I'll have a look at least to see if the `replicate_to_region` and `wait_for_image` functions  can be carved out to some utils file, so the code is not repeated.

File: mash/services/test_preparation_service.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ def main():
         job_factory = BaseJobFactory(
             service_name=service_name,
             job_types={
+                'ec2': EC2TestPreparationJob
```
[the remark this answers is not in the document]
Comment (apozsuse) [reply mash/services/test_preparation_service.py:47]: Missed that part. I'll include it. Thx!

File: mash/services/jobcreator/ec2_job.py
[the remark this answers is not in the document]
Comment (apozsuse) [reply mash/services/jobcreator/ec2_job.py:1]: Missed that part. I'll include it.

**Comment by smarlowucf** [comment]
> There's an option to switch the order of the replicate service and put it after create, but the clean up if something fails would be more complex.

I was thinking of the idea of having the service use the existing replicate classes? Not sure if it'd be possible though

**Comment by apozsuse** [comment]
> > There's an option to switch the order of the replicate service and put it after create, but the clean up if something fails would be more complex.
> 
> I was thinking of the idea of having the service use the existing replicate classes? Not sure if it'd be possible though

Checked and it's possible to reuse the class, leading to a solution with much less code duplication.
I'll close this PR and create a new one with the new approach.

**Pull request closed without merging**